ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/composables/TierSwitcher.kt

```diff
@@ -0,0 +1,132 @@
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.onSizeChanged
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import com.revenuecat.purchases.ui.revenuecatui.data.processed.TemplateConfiguration
+
+private object TierSwitcherUIConstants {
+    const val roundedCorner = 50
+    val height = 40.dp
+    val selectedTierPadding = 2.dp
+}
+
+@Composable
+@Suppress("LongParameterList")
+internal fun SelectedTierView(
+    selectedTier: TemplateConfiguration.TierInfo,
+    backgroundSelectedColor: Color,
+    foregroundSelectedColor: Color,
+) {
+    Box(
+        modifier = Modifier
+            .background(
+                color = backgroundSelectedColor,
+                shape = RoundedCornerShape(TierSwitcherUIConstants.roundedCorner),
+            )
+            .padding(horizontal = 16.dp, vertical = 8.dp),
+    ) {
+        Text(
+            text = selectedTier.name,
+            color = foregroundSelectedColor,
+            style = MaterialTheme.typography.bodyMedium,
+        )
+    }
+}
+
+@Composable
+@Suppress("LongParameterList")
+internal fun TierSwitcher(
+    tiers: List<TemplateConfiguration.TierInfo>,
+    selectedTier: TemplateConfiguration.TierInfo,
+    onTierSelected: (TemplateConfiguration.TierInfo) -> Unit,
+    backgroundColor: Color,
+    backgroundSelectedColor: Color,
+    foregroundColor: Color,
+    foregroundSelectedColor: Color,
+) {
+    val selectedIndex = tiers.indexOf(selectedTier)
+    var totalWidthPx by remember { mutableStateOf(0) }
+
+    val density = LocalDensity.current
+
+    Box(
+        Modifier
+            .padding(TierSwitcherUIConstants.selectedTierPadding)
+            .clip(RoundedCornerShape(TierSwitcherUIConstants.roundedCorner))
+            .background(backgroundColor)
+            .height(TierSwitcherUIConstants.height)
+            .fillMaxWidth()
+            .onSizeChanged { size ->
+                totalWidthPx = size.width
+            },
+    ) {
+        val optionWidth = with(density) { (totalWidthPx / tiers.size).toDp() }
+        val indicatorOffset by animateDpAsState(targetValue = optionWidth * selectedIndex, label = "tier_switcher")
+
+        Box(
+            Modifier
+                .offset(x = indicatorOffset)
+                .fillMaxHeight()
+                .width(optionWidth)
```

Comment:
Hmm just to check, have you tried assigning a percentage of the available size to the size here? So having this be `fillMaxWidth(1f/tiers.size)`. Not sure if that would work, but it would avoid the recomposition when the size is calculated.

Reply:
Seems like this works! 😊